LoopKit/GlucoseRangeSchedule.swift

```diff
@@ -57,8 +57,45 @@ extension DoubleRange: Hashable {}
 public struct GlucoseRangeSchedule: DailySchedule, Equatable {
     public typealias RawValue = [String: Any]
 
+    /// A time-based value overriding the rangeSchedule
+    public struct Override: Equatable {
+
+        public let start: Date
+        public let end: Date
+        public let value: DoubleRange
+
+        /// Initializes a new override
+        ///
+        /// - Parameters:
+        ///   - start: The date at which the override starts
+        ///   - end: The date at which the override ends, or nil for an indefinite override
+        ///   - value: The value to return when active
+        public init(start: Date, end: Date?, value: DoubleRange) {
```

Comment:
[nit] consider `end: Date? = nil` so that caller can leave out `end` parameter altogether.